Formula/elasticsearch@7.6.rb

```diff
@@ -0,0 +1,150 @@
+    # Make sure runtime directories exist
+    (var/"lib/elasticsearch").mkpath
+    (var/"log/elasticsearch").mkpath
+    ln_s etc/"elasticsearch", libexec/"config" unless (libexec/"config").exist?
+    (var/"elasticsearch/plugins").mkpath
+    ln_s var/"elasticsearch/plugins", libexec/"plugins" unless (libexec/"plugins").exist?
+    # fix test not being able to create keystore because of sandbox permissions
+    system bin/"elasticsearch-keystore", "create" unless (etc/"elasticsearch/elasticsearch.keystore").exist?
+  end
+
+  def caveats
+    s = <<~EOS
+      Data:    #{var}/lib/elasticsearch/
+      Logs:    #{var}/log/elasticsearch/#{cluster_name}.log
+      Plugins: #{var}/elasticsearch/plugins/
+      Config:  #{etc}/elasticsearch/
+    EOS
+
+    s
+  end
+
+  plist_options :manual => "elasticsearch"
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>KeepAlive</key>
+          <false/>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{opt_bin}/elasticsearch</string>
+          </array>
+          <key>EnvironmentVariables</key>
+          <dict>
+          </dict>
+          <key>RunAtLoad</key>
+          <true/>
+          <key>WorkingDirectory</key>
+          <string>#{var}</string>
+          <key>StandardErrorPath</key>
+          <string>#{var}/log/elasticsearch.log</string>
+          <key>StandardOutPath</key>
+          <string>#{var}/log/elasticsearch.log</string>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    port = free_port
+    system "#{bin}/elasticsearch-plugin", "list"
+    pid = testpath/"pid"
+    begin
+      system "#{bin}/elasticsearch", "-d", "-p", pid,
+             "-Epath.data=#{testpath}/data",
+             "-Epath.logs=#{testpath}/logs",
+             "-Ehttp.port=#{port}"
+      sleep 10
+      output = shell_output("curl -s -XGET localhost:#{port}/")
+      assert_equal "oss", JSON.parse(output)["version"]["build_flavor"]
+    ensure
+      Process.kill(9, pid.read.to_i)
+    end
+
+    port = free_port
+    (testpath/"config/elasticsearch.yml").write <<~EOS
+      path.data: #{testpath}/data
+      path.logs: #{testpath}/logs
+      node.name: test-es-path-conf
+      http.port: #{port}
+    EOS
+
+    cp etc/"elasticsearch/jvm.options", "config"
+    cp etc/"elasticsearch/log4j2.properties", "config"
+
+    ENV["ES_PATH_CONF"] = testpath/"config"
+    pid = testpath/"pid"
+    begin
+      system "#{bin}/elasticsearch", "-d", "-p", pid
+      sleep 10
+      system "curl", "-XGET", "localhost:#{port}/"
+      output = shell_output("curl -s -XGET localhost:#{port}/_cat/nodes")
+      assert_match "test-es-path-conf", output
+    ensure
+      Process.kill(9, pid.read.to_i)
+    end
```

Comment:
I can't tell the difference between this and the other one. What are you testing additionally here?

Reply:
The 2nd one tests a config file (not sure if two are really needed, but it's in the current version).